### PR TITLE
Feature/vsphere disk resizing

### PR DESCRIPTION
**Pull request by ycliuhw**
### Checklist

 - [x] Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change?
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR?
 - [x] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed?
 - [x] Do comments answer the question of why design decisions were made?

----

## Description of change

Changed to use `Reconfigure` API for disk resizing;

## QA steps

```console
# bootstrap to vSphere
# then add-machine
juju add-machine --constraints root-disk=25G

# ssh into the machine to check disk size
df -h
```

## Documentation changes

No(Internal change);

## Bug reference

No

### CODE REVIEW

**Comment by ycliuhw** [comment]
[vSphere resizing disk](https://trello.com/c/oqPX4s7F/1026-vsphere-resizing-disk)

File: provider/vsphere/internal/vsphereclient/createvm.go [hunk 1]
```diff
@@ -398,7 +398,7 @@ func (c *Client) extendVMRootDisk(
 	sizeMB uint64,
 	taskWaiter *taskWaiter,
 ) error {
-	disk, backing, err := c.getDiskWithFileBacking(ctx, vm)
+	disk, _, err := c.getDiskWithFileBacking(ctx, vm)
```
Comment (babbageclunk) [inline provider/vsphere/internal/vsphereclient/createvm.go]: Might as well change this method to `getDisk` and drop the backing return value, it doesn't look like anything else uses it.

**Comment by ycliuhw** [comment]
!!build!!

**Comment by ycliuhw** [comment]
!!build!!

**Comment by ycliuhw** [comment]
$$merge$$

**Pull request merged**